onnxruntime/core/mlas/lib/sqnbitgemm_kernel_avx2_int8_blklen32.h

```diff
@@ -1660,7 +1660,24 @@ MlasQ4Int8TileGemmKernelBlkLen32Avx2(
 
             if constexpr (NCols4 == 8) {
                 __m128 acc_0 = FoldAccumulators(acc[0], acc[1], acc[2], acc[3]);
+
```

Comment:
Or you rewrite the code to remove the if constexpr usages? I think the current code is a little bit complicated that clang does not know this branch cannot be reached.

Reply:
The whole `MlasQ4Int8TileGemmKernelBlkLen32Avx2` function seems not used anywhere. I try to keep the code there for future reference.